xbmc/GUIPassword.h

```diff
@@ -25,6 +25,7 @@ class CGUIPassword : public ISettingCallback
 public:
   CGUIPassword(void);
   ~CGUIPassword(void) override;
+  template <typename T> bool IsItemUnlocked(T pItem, const std::string& strType, const std::string& strLabel, const std::string& strHeading);
   bool IsItemUnlocked(CFileItem* pItem, const std::string &strType);
   bool IsItemUnlocked(CMediaSource* pItem, const std::string &strType);
```

Comment:
I'm confused, was this function not implemented and dangling in the header before now?

Reply:
That's correct, the overload for cmediasource isn't implemented

Reply:
Right, sorry guys i pushed something new, now it's perfect:)

Reply:
got it. Can you move the docstrings from the functions into the headers? Separation of interface and implementation.

Reply:
**GUIPassword.cpp**:
`.....
bool CGUIPassword::IsItemUnlocked(CFileItem* pItem, const std::string &strType)
{  const s.....`

**GUIPassword.h**:
`bool IsItemUnlocked(CFileItem* pItem, const std::string &strType);`

they're identical for this example. what should i move?

Reply:
The documentation *inside* the function in the .cpp should be moved to the header

Reply:
ok, now i got, will come soon